coursera/api.py

```diff
@@ -543,8 +544,15 @@ def extract_links_from_lecture(self,
         if assets is None:
             assets = []
 
-        links = self._extract_videos_and_subtitles_from_lecture(
-            video_id, subtitle_language, resolution)
+        try:
+            links = self._extract_videos_and_subtitles_from_lecture(
+                video_id, subtitle_language, resolution)
+        except HTTPError as ex:
+            FORBIDDEN = 403
+            if ex.response.status_code == FORBIDDEN:
```

Comment:
A great and really useful addition to this pull request would be a research on why the item is locked. Ideally, we would like to tell user the reason. For example, "The course <name> contains locked items because its session has already ended. Switch to new session here <URL> to unlock the items."